actors/market/src/lib.rs

```diff
@@ -367,8 +359,14 @@ impl Actor {
 
                 // Store verified allocation (if any) in the pending allocation IDs map.
                 // It will be removed when the deal is activated or expires.
-                if valid_deal.allocation != NO_ALLOCATION_ID {
-                    pending_deal_allocation_ids.push((deal_id_key(deal_id), valid_deal.allocation));
+                if valid_deal.proposal.verified_deal {
+                    pending_deal_allocation_ids.push((
+                        deal_id_key(deal_id),
+                        client_allocations
+                            .get_mut(&valid_deal.proposal.client.id().unwrap())
+                            .unwrap()
+                            .remove(0),
```

Comment:
This relies on the parallel nature of the allocation IDs and valid deals lists, which is fragile. I have some memory of being unsettled by this earlier, which you resolved, but now my feedback has led it to return.

I will have a go trying to structure this myself.

Reply:
thanks for your review, what should i do now?

Reply:
@swift-mx Let's refine the patch after anorth's structure done.

Reply:
ok, thanks